safenode/src/bin/kadnode.rs

```diff
@@ -1,233 +1,178 @@
-            let mdns = mdns::async_io::Behaviour::new(mdns::Config::default(), local_peer_id)?;
-            let behaviour = MyBehaviour { kademlia, mdns };
-
-            let mut swarm =
-                SwarmBuilder::with_async_std_executor(transport, behaviour, local_peer_id).build();
-
-            // // Listen on all interfaces and whatever port the OS assigns.
-            let addr = "/ip4/0.0.0.0/udp/0/quic-v1".parse().expect("addr okay");
-            swarm.listen_on(addr).expect("listening failed");
-
-            swarm
-        };
-
-        let net_info = swarm.network_info();
-
-        debug!("network info: {net_info:?}");
-        // Kick it off.
+#[async_std::main]
+async fn main() -> Result<()> {
+    let opt = Opt::parse();
+    let _log_appender_guard = init_node_logging(&opt.log_dir)?;
+
+    let (mut network_api, mut network_events, network_event_loop) = EventLoop::new()?;
+    let temp_dir = TempDir::new()?;
+    let storage = DataStorage::new(&temp_dir);
+
+    // Spawn the network task for it to run in the background.
+    spawn(network_event_loop.run());
+
+    // Todo: Implement node bootstrapping to connect to peers from outside local network
+    // In case a listen address was provided use it, otherwise listen on any
+    // address.
+    // match opt.listen_adress {
+    //     Some(addr) => network_client
+    //         .start_listening(addr)
+    //         .await
+    //         .expect("Listening not to fail."),
+    //     None => network_client
+    //         .start_listening("/ip4/0.0.0.0/tcp/0".parse()?)
+    //         .await
+    //         .expect("Listening not to fail."),
+    // };
+
+    // In case the user provided an address of a peer on the CLI, dial it.
+    // if let Some(addr) = opt.peer {
+    //     let peer_id = match addr.iter().last() {
+    //         Some(Protocol::P2p(hash)) => PeerId::from_multihash(hash).expect("Valid hash."),
+    //         _ => return Err("Expect peer multiaddr to contain peer ID.".into()),
+    //     };
+    //     network_client
+    //         .dial(peer_id, addr)
+    //         .await
+    //         .expect("Dial to succeed");
+    // }
+    //
+
+    let mut api_clone = network_api.clone();
+    let storage_clone = storage.clone();
+    let (peer_dicovered_send, peer_dicovered_rx) = oneshot::channel();
+    let mut peer_dicovered_send = Some(peer_dicovered_send);
+    spawn(async move {
         loop {
-            select! {
-                cmd = receiver.recv().fuse() => {
-                    debug!("Cmd in: {cmd:?}");
-                    if let Some(SwarmCmd::Search(xor_name)) =  cmd {
-                        swarm.behaviour_mut().get_closest_peers_to_xorname(xor_name);
+            let event = match network_events.next().await {
+                Some(event) => event,
+                None => continue,
+            };
+            match event {
+                NetworkEvent::InboundSafeRequest { req, channel } => {
```

Comment:
A small note: Long time ago we cleared all references to `Safe` in the safe_network types, fns and varibles naming.. It was deemed superfluous. I.e. `InboundSafeRequest` would be better without `Safe`.